web-marketplace/src/components/organisms/Order/Order.Credits.tsx

```diff
@@ -0,0 +1,66 @@
+import { msg } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+
+import { DenomIconWithCurrency } from 'web-components/src/components/DenomIconWithCurrency/DenomIconWithCurrency';
+import { SupCurrencyAndAmount } from 'web-components/src/components/SupCurrencyAndAmount/SupCurrencyAndAmount';
+import { Title } from 'web-components/src/components/typography';
+
+import { OrderDetailsIcon } from './Order.Icon';
+import { OrderSummaryRow } from './Order.SummaryRow';
+import { CreditsData, OrderSummarySectionProps } from './Order.types';
+
+export const OrderCredits = ({
+  icon,
+  title,
+  data,
+}: OrderSummarySectionProps) => {
+  const { _ } = useLingui();
+  const { credits, price, denom } = data as CreditsData;
+  const totalPrice = +credits * +price;
+  return (
+    <section className="flex gap-20">
```

Comment:
In this case, I think creating a reusable component for these sections may add unnecessary complexity because the logic for each component is slightly different. While they share some structure (title, icon, rows), combining them into a single component would require a few conditional checks that would make them harder to work with. I believe that keeping them separated while using shared internal components allows each component to be simple to maintain and update.

Reply:
then what about just having a reusable component with title, icon and children, which makes it more flexible?